domain/controllernode/service/service.go

```diff
@@ -259,12 +259,51 @@ func (s *Service) GetAPIAddresses(ctx context.Context, nodeID string) ([]string,
 	return s.st.GetAPIAddresses(ctx, nodeID)
 }
 
-// GetAllAPIAddressesForAgents returns a map of controller IDs to their API
+// GetAPIHostPortsByControllerIDForAgents returns API HostPorts that are available
+// for agents. The map is keyed by controller ID, and the values are HostPorts
+// representing the API addresses for each controller node.
+func (s *Service) GetAPIHostPortsByControllerIDForAgents(ctx context.Context) ([]network.HostPorts, error) {
```

Comment:
There's no keying by controller or controller argumentation, making this a misleading name. Comment needs updating. Applies to method below too.